libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4029,11 +4029,25 @@ void TypeChecker::endVisit(Literal const& _literal)
 			suffixFunctionType->hasBoundFirstArgument() ||
 			!_literal.suffixFunction()->usableAsSuffix()
 		)
+		{
+			auto suffixDefinitionLocation = SecondarySourceLocation{};
+			if (
+				suffixFunctionType &&
+				suffixFunctionType->hasDeclaration() &&
+				dynamic_cast<Declaration const*>(&suffixFunctionType->declaration())
+			)
+				suffixDefinitionLocation.append(
+					"Invalid suffix function defined here:",
```

Comment:
Since it might not be a function, I'd avoid saying that in the message:
```suggestion
					"Suffix defined here:",
```